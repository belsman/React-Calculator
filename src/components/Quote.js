import React from 'react';

const Quote = () => (
  <div className="content">
    <p
      className="quotes"
      style={{
        top: '50%',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        position: 'absolute',
        textAlign: 'center',
        fontSize: '1.3rem',
      }}
    >
      Mathematics is about numbers, equations, computations, or
      algorithms: it is about understanding. -- William Paul Thurston
    </p>
  </div>
);

export default Quote;
