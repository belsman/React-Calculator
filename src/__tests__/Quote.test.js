import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Quote from '../components/Quote';

describe('Quote component test', () => {
  test('should renders the corrent quote', () => {
    render(<Quote />);
    const text = screen.getByText(/Mathematics is about numbers,/i);
    expect(text).toBeInTheDocument();
  });

  test('should render the expected output', () => {
    const component = renderer.create(
      <Quote />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
