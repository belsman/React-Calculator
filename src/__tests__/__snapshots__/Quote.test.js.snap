// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component test should render the expected output 1`] = `
<div
  className="content"
>
  <p
    className="quotes"
    style={
      Object {
        "fontSize": "1.3rem",
        "left": "50%",
        "position": "absolute",
        "textAlign": "center",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      }
    }
  >
    Mathematics is about numbers, equations, computations, or algorithms: it is about understanding. -- William Paul Thurston
  </p>
</div>
`;
